book/src/qobject/generated-qobject.md

```diff
@@ -0,0 +1,112 @@
+
+## Referencing another QObject
+
+The `qobject::T` type is defined both within the CXX-Qt bridge, as well as the surrounding module.
+The simplest way is to reference it from the surrounding module.
+
+Example:
+``` rust,ignore,noplayground
+// In file qt_types.rs
+#[cxx_qt::bridge]
+mod ffi {
+  #[cxx_qt::qobject]
+  #[derive(Default)]
+  pub struct MyObject {}
+}
+```
+
+``` rust,ignore,noplayground
+// In another module
+fn my_function(parameter: &crate::qt_types::qobject::MyObject) { /* ... */ }
+```
+
+Unfortunately, nested QObjects aren't yet supported by CXX-Qt.
+Progress can be tracked in [#299](https://github.com/KDAB/cxx-qt/issues/299).
+
+## `impl qobject::T`
+As mentioned before, the C++ QObject is exposed to Rust as an opaque CXX type.
+This allows you to implement methods for it in Rust using normal `impl` blocks.
+Because `qobject::T` is an opaque C++ type, the same rules as with normal CXX opaque types apply.
+Most importantly this means that the type may never be accessed by value or by mutable self reference directly.
+Rather, all methods must use either `&self` or `self: Pin<&mut Self>` as the self types.
+
+For more information about pinning, see: https://doc.rust-lang.org/std/pin/
+
+In addition to methods that extend the Rust interface of a `qobject::T`, you may also mark methods within an `impl qobject::T` block with the `#[qinvokable]` attribute.
+These methods will then be exposed to the C++ QObject and can then be called by C++ or QML.
+See the [QObject page](./qobject_struct.md#invokables) for more details.
+
+## Available Functions
+
+Every `qobject::T` struct provides the following methods:
+
+### Access to the Qt thread
+``` rust,ignore,noplayground
+fn qt_thread(&self) -> UniquePtr<CxxQtThread>
+```
+This function provides you with a handle to the Qt thread that the QObject resides in.
+This is helpful as the QObject itself is neither Send, nor Sync.
+The CxxQtThread however is Send and can therefore be moved into a different thread.
+By using the `queue` function on the CxxQtThread, you may then queue a Rust lambda onto the Qt thread again.
+The lambda also takes a pinned mutable reference to the QObject, so that it can modify it.
+
+See the [CxxQtThread page](./cxxqtthread.md) for more details.
+
+### Signal emission
+``` rust,ignore,noplayground
+fn emit(self: Pin<&mut Self>, signal: /*Your Signals enum goes here*/)
+```
+If there is a [Signals enum](./signals_enum.md) defined, CXX-Qt will generate the appropriate `emit` function to allow you to emit signals.
+
+See the [Signals enum page](./signals_enum.md) for more details.
+
+### Access to internal Rust struct
+For every field in the Rust struct, CXX-Qt will generate appropriate getters and setters for every field in the Rust struct.
+See the [QObject page](./qobject_struct.md#properties) for details.
+
+There is also an advanced way to access the data in the internal Rust struct:
```

Comment:
```suggestion
There is also a direct way to access the data in the internal Rust struct:
```

Reply:
I'd prefer to keep the "advanced" wording here.
Calling the accessors is pretty much the direct way to access the data.
The `rust()` and `rust_must()` methods add an additional call that you're only supposed to use for advanced purposes, e.g. if you want to modify a property without completely setting a new value.
You shouldn't need this for >90% of cases.

Reply:
I agree, you should primarily be using `prop` `prop_mut` `set_prop`. This is for advanced users, eg when you might want to modify multiple fields at once instead of using `prop_mut` etc.

Reply:
sure :shrug: